app/src/main/java/in/tosc/digitaloceanapp/activities/DetailDropletActivity.java

```diff
@@ -100,6 +104,40 @@ public boolean onOptionsItemSelected(MenuItem item) {
         if (id == R.id.delete_droplet) {
 
             // TODO: 26/11/16 perform delete
+            new MaterialDialog.Builder(this)
+                    .title(R.string.delete_droplet)
+                    .content(getString(R.string.dialog_delete_droplet_msg,droplet.getName()))
+                    .positiveText(R.string.dialog_confirm).onPositive(new MaterialDialog.SingleButtonCallback() {
+                @Override
+                public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
+                    Snackbar.make(coordinatorLayout, getString(R.string.delete_droplet_msg), Snackbar.LENGTH_INDEFINITE).show();
+                    doaClient.performAction(droplet.getId(),ActionType.DESTROY,null).enqueue(new Callback<Action>() {
+                        @Override
+                        public void onResponse(Call<Action> call, Response<Action> response) {
+                            Log.d("DESTROY",String.valueOf(response.code()));
+                            new Handler().postDelayed(new Runnable() {
+                                @Override
+                                public void run() {
+                                    DropletActivity.refreshData();
+                                    finish();
+                                }
+                            },getResources().getInteger(R.integer.REFRESH_AFTER_DELETE_TIMEOUT));
```

Comment:
No, what I meant was, why are you waiting for a fixed time, you should only wait for the time which is actually taken while deleting this droplet.